op-deployer/pkg/deployer/bootstrap/holocene.go

```diff
@@ -0,0 +1,292 @@
+
+	result, err = loader.CallContract(ctx, l1Client, snapshots.LoadDisputeGameFactoryABI(), dgfAddr, "gameImpls", uint32(0))
+	if err != nil {
+		return fmt.Errorf("failed to load current FaultDisputeGame implementation addr: %w", err)
+	}
+	cannonGameImpl := result.GetAddress(0)
+	lgr.Info("found FaultDisputeGame", "addr", cannonGameImpl)
+
+	result, err = loader.CallContract(ctx, l1Client, snapshots.LoadDisputeGameFactoryABI(), dgfAddr, "gameImpls", uint32(1))
+	if err != nil {
+		return fmt.Errorf("failed to load current PermissionedDisputeGame implementation addr: %w", err)
+	}
+	permissionedGameImpl := result.GetAddress(0)
+	lgr.Info("found PermissionedDisputeGame", "addr", permissionedGameImpl)
+
+	dgfABI := snapshots.LoadFaultDisputeGameABI()
+	result, err = loader.CallContract(ctx, l1Client, dgfABI, permissionedGameImpl, "vm")
+	if err != nil {
+		return fmt.Errorf("failed to load current MIPS implementation address: %w", err)
+	}
+	oldMIPS := result.GetAddress(0)
+	lgr.Info("found existing MIPS", "addr", oldMIPS)
+
+	result, err = loader.CallContract(ctx, l1Client, snapshots.LoadMIPSABI(), oldMIPS, "oracle")
+	if err != nil {
+		return fmt.Errorf("failed to load PreimageOracle address: %w", err)
+	}
+	oracleAddr := result.GetAddress(0)
+	lgr.Info("found PreimageOracle", "addr", oracleAddr)
+
+	lgr.Info("using absolute prestate", "hash", cfg.AbsolutePrestate)
+
+	deployment := &DeployHoloceneOutput{}
+
+	// First deploy the update MIPS contract
+	mipsDeployment, err := opcm.DeployMIPS(
+		host,
+		opcm.DeployMIPSInput{
+			MipsVersion:    1,
+			PreimageOracle: oracleAddr,
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("error deploying dispute game: %w", err)
+	}
+
+	lgr.Info("deployed new mips", "addr", mipsDeployment.MipsSingleton)
+	deployment.MipsSingleton = mipsDeployment.MipsSingleton
+
+	// Populate required code in the local state
+	addresses := []common.Address{
+		oracleAddr,
+	}
+	for _, addr := range addresses {
+		code, err := l1Client.CodeAt(ctx, addr, nil)
+		if err != nil {
+			return fmt.Errorf("failed to get code for %v: %w", addr, err)
+		}
+		host.ImportAccount(addr, types.Account{
+			Code: code,
+		})
+	}
+
+	// Next deploy new FaultDisputeGame if one was already present
+	if cannonGameImpl != (common.Address{}) {
+		out, err := deployDisputeGame(ctx, l1Client, cannonGameImpl, false, host, mipsDeployment.MipsSingleton, cfg.AbsolutePrestate)
+		if err != nil {
+			return err
+		}
+		lgr.Info("deployed FaultDisputeGame", "addr", out.DisputeGameImpl)
+		deployment.FaultDisputeGame = out.DisputeGameImpl
+	}
+
+	// Deploy PermissionedDisputeGame
+	out, err := deployDisputeGame(ctx, l1Client, permissionedGameImpl, true, host, mipsDeployment.MipsSingleton, cfg.AbsolutePrestate)
+	if err != nil {
+		return err
+	}
+	lgr.Info("deployed PermissionedDisputeGame", "addr", out.DisputeGameImpl)
+	deployment.PermissionedDisputeGame = out.DisputeGameImpl
+
+	// TODO: Deploy the updated SystemConfig?
```

Comment:
cc @sebastianst @geoknee it looks like this is where you plug into `op-deployer`